packages/dds/sequence/src/test/intervalCollection.spec.ts

```diff
@@ -1389,4 +1425,40 @@ describe("SharedString interval collections", () => {
 			assertIntervals(sharedString, collection1, [{ start: 5, end: 8 }]);
 		});
 	});
+
+	describe("querying intervals with index API's", () => {
+		it("can attach/detach an index", () => {
+			sharedString.initializeLocal();
+			const collection = sharedString.getIntervalCollection("test");
+			sharedString.insertText(0, "xyzabc");
+			const id1 = collection.add(1, 1, IntervalType.SlideOnRemove).getIntervalId();
+			const id2 = collection.add(1, 3, IntervalType.SlideOnRemove).getIntervalId();
+
+			const mockIntervalIndex = new MockIntervalIndex();
+			collection.attachIndex(mockIntervalIndex);
+
+			// Verify if all intervals in this collection are added to the attached index
+			assert.strictEqual(collection.getIntervalById(id1), mockIntervalIndex.get(0));
+			assert.strictEqual(collection.getIntervalById(id2), mockIntervalIndex.get(1));
+
+			// Verify if the intervals in collection are synced with those in collection
+			const id3 = collection.add(2, 5, IntervalType.SlideOnRemove).getIntervalId();
```

Comment:
i would split these assertions into a separate test case. In general, always try to keep your test cases minimal. This has the benefit of having some idea of what went wrong knowing only the test name.